### PR TITLE
Option 1: App Generator suspenders style

### PR DESCRIPTION
**Pull request by drumusician**
An initial setup of creating a greenfield app with
launch_base to hit the ground running.

This boilerplate does the following:
* Inherit from the Rails AppGenerator and customise a few things
* Throw in a custom Gemfile
* Update the .ruby-version file with our version

===
* Note that the Gemfile is just something I threw in for now. I couldn't find what was decided in the card... ?
* This is option 1. I feel it might be better to use Rails Templates instead. The gives us a much more concise DSL. We could then wrap that up in our gem so it feels like our gem is the generator but under the hood it will use Rails templates.

### CODE REVIEW

File: bin/launch_base [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+source_path = (Pathname.new(__FILE__).dirname + '../lib').expand_path
+$LOAD_PATH << source_path
+
+require 'launch_base'
+
+if ARGV.empty?
+  puts 'Please provide a path for the new application'
+  exit 0
+elsif ['-v', '--version'].include? ARGV[0]
```
Comment (nickve28) [inline bin/launch_base]: wouldnt it be more like

```ARGV.any? { |arg| arg == '-v' || arg == '--version' }``` ?

File: lib/launch_base/app_builder.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module LaunchBase
+  # Customise the Rails AppBuilder
```
Comment (nickve28) [inline lib/launch_base/app_builder.rb]: Customize*

File: lib/launch_base/app_builder.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    end
+
+    def set_ruby_to_version_being_used
+      create_file '.ruby-version', "#{LaunchBase::RUBY_VERSION}\n", force: true
+    end
+
+    def configure_generators
+      config = <<-RUBY
+
+      config.generators do |generate|
```
Comment (nickve28) [inline lib/launch_base/app_builder.rb]: this should be indented for clarify I guess

File: lib/launch_base/version.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
 module LaunchBase
   VERSION = '0.1.0'.freeze
+  RAILS_VERSION = '~> 5.2.0'.freeze
+  RUBY_VERSION = '2.5.1'.freeze
```
Comment (nickve28) [inline lib/launch_base/version.rb]: ❄️

File: templates/Gemfile.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+source 'https://rubygems.org'
+
+gem 'rails', '<%= LaunchBase::RAILS_VERSION %>'
+
+gem 'pg'
+gem 'puma', '~> 3.11'
+gem 'sass-rails', '~> 5.0'
+gem 'uglifier', '>= 1.3.0'
+gem 'turbolinks', '~> 5'
+gem 'jbuilder', '~> 2.5'
```
Comment (nickve28) [inline templates/Gemfile.erb]: is this needed?

File: templates/Gemfile.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+gem 'bootsnap', '>= 1.1.0', require: false
+
+group :development, :test do
+  gem 'pry'
```
Comment (nickve28) [inline templates/Gemfile.erb]: I think byebug is preferred nowadays right?

File: lib/launch_base/app_builder.rb (hunk 2 above)
Comment (drumusician) [reply lib/launch_base/app_builder.rb]: English vs American:)

File: templates/Gemfile.erb (hunk 5 above)
Comment (drumusician) [reply templates/Gemfile.erb]: no we need to decide what goes in the Gemfile, this is just the basic Rails stuff

File: templates/Gemfile.erb (hunk 6 above)
Comment (drumusician) [reply templates/Gemfile.erb]: up for discussion I guess. According to @martijnversluis byebug doesn't support multiple threads

File: templates/Gemfile.erb (hunk 5 above)
Comment (nickve28) [reply templates/Gemfile.erb]: but I don't think we use jbuilder that much? @kabisa/kabisa-amsterdam

File: templates/Gemfile.erb [hunk 7]
```diff
@@ -0,0 +1,29 @@
+source 'https://rubygems.org'
+
+gem 'rails', '<%= LaunchBase::RAILS_VERSION %>'
+
+gem 'pg'
+gem 'puma', '~> 3.11'
+gem 'sass-rails', '~> 5.0'
+gem 'uglifier', '>= 1.3.0'
+gem 'turbolinks', '~> 5'
```
Comment (nickve28) [inline templates/Gemfile.erb]: idem for this one

**Comment by drumusician** [comment]
after discussing the 2 solutions we have decided to use the Rails Template API so closing this pull request

**Pull request closed without merging**